### DISCUSSION TITLE
Photon 3D LUT Editor / Color Grading Tool

### DISCUSSION THREAD

**Discussion by PirminStraub** [ocornut/imgui#3792] [Old/Archived discussions]
Looks like we will be the ones to kick off the Show&Tell section 🙌 

[Photon 3D](https://www.color.io/photon) is a Color Grading Tool / LUT editor that allows to change the look of an image or video by interacting directly with the 3D color model. This means that geometric operations on the color model translate to color changes in the image.

![Overview_Operate_Spindel](https://user-images.githubusercontent.com/42964073/107032488-34a5fe00-67b4-11eb-8381-3a263e64e358.gif)

These operations are achieved by combining custom tools and ranges to achieve the desired effects. For example higher contrast (tool) in a certain hue angle (range). Or more saturation (tool) over the entire image (range).

Color operations can also be carried out by clicking and dragging directly on the image.

![Photon_PaintOnImage](https://user-images.githubusercontent.com/42964073/107039001-a46cb680-67bd-11eb-9904-575216323abe.gif)

Photon 3D can load and edit pre-existing Lookup Tables or you can create your own looks from scratch. The finished LUTs can be exported in the .cube format that can be (re)used across all major image and video editing software. Of course images can be exported as well. Photon 3D also comes with an OpenFX Connector Plugin for DaVinci Resolve to work on video projects.

It features a File Library to manage and preview LUTs and Images, as well as several tools to smooth, combine and mix different looks, a history timeline to dial through the edits and a snapshot feature which serves as sort of a quick save / memory bank.

![Media Library](https://user-images.githubusercontent.com/42964073/107039132-c2d2b200-67bd-11eb-925b-0492e1ffdde1.gif)

We started using Dear ImGui early on in the research and development process and we really enjoyed working with it so we decided to keep it for our transition into the final product.

We built our own GUI elements such as animated buttons and the Snapshot mixer on top of what Dear ImGui offered us and also a couple of helper functions for layout, styling etc.

![Snapshot Mixer](https://user-images.githubusercontent.com/42964073/107039227-e1d14400-67bd-11eb-9432-768791cfc0d7.gif)

We're happy to get your feedback and if you are interested in trying out Photon 3D for yourself, head on over to www.color.io/photon - there is a free trial version available 🚀 

Thanks ocornut and the Dear ImGui team for your consistent and hard work, we really appreciate it!

All the best from Cologne, Germany.
Pirmin and Manu

**Comment by TheEnbyWitch** [comment]
Wow, this looks so cool!!!

**Reply by TheEnbyWitch** [reply]
BTW, question, if were to buy the software (without the OFX plugin), would I still have access to it after a year?

**Reply by PirminStraub** [reply]
Hi Luna, glad we caught your interest 😊 ~~When your Photon License expires, the application will fall back into free trial mode. The application will continue to work but you won't be able to export LUTs and images. Would you still be interested?~~ **Edit:** Licenses are now permanent. We hope you enjoy Photon 🚀

**Reply by DGriffin91** [reply]
@PirminStraub Maybe it's just me, but "1 Year Updates and Support" seems more to imply that just updates and support would cease to function after 1 year, not that the whole program goes into trial mode. You may want to consider re-wording on the website. I was considering buying it until I saw this comment, and am glad I didn't buy it not knowing it would stop working after a year.

This does look super cool. Great work on the overall design.

**Reply by PirminStraub** [reply]
You're right, we rephrased it to "1 year Export, Updates and Support" to make it more clear. Thanks for pointing this out to us! We hope you reconsider. And thanks for your appreciation of the overall design 😊

**Reply by PirminStraub** [reply]
@LunaRyuko @DGriffin91 - just to let you know, we have updated our purchase options to permanent licenses - you were not the only ones questioning our previous pricing / subscription model 😅 in case you already bought a license, you will be upgraded automatically to a permanent one.

**Comment by hb3p8** [comment]
Great work! Cool to see non-standard Dear Imgui solutions. Love how elegantly you handled the tutorial text layout!

PS Guess it's time to implement cube LUT support in Light Tracer Render 😄.

**Reply by PirminStraub** [reply]
shoot me a message - we might have some code snippets sitting around to make it happen 😄

**Reply by hb3p8** [reply]
I'd like to connect but I don't see how 😄 You can find me on twitter btw (@hb3p8)

**Comment by alexhajdu** [comment]
Impressive!

**Comment by gpakosz** [comment]
@PirminStraub how did you bind dearimgui?

**Reply by PirminStraub** [reply]
GLFW
